src/main/java/com/openshift/jenkins/plugins/pipeline/dsl/OpenShiftExec.java

```diff
@@ -0,0 +1,196 @@
+package com.openshift.jenkins.plugins.pipeline.dsl;
+
+import com.openshift.jenkins.plugins.pipeline.Argument;
+import com.openshift.jenkins.plugins.pipeline.ParamVerify;
+import com.openshift.jenkins.plugins.pipeline.model.GlobalConfig;
+import com.openshift.jenkins.plugins.pipeline.model.IOpenShiftExec;
+import hudson.Extension;
+import hudson.model.AbstractBuild;
+import hudson.model.AbstractProject;
+import hudson.model.Action;
+import hudson.model.BuildListener;
+import hudson.tasks.BuildStepMonitor;
+import org.jenkinsci.plugins.workflow.steps.AbstractStepDescriptorImpl;
+import org.jenkinsci.plugins.workflow.steps.Step;
+import org.kohsuke.stapler.DataBoundConstructor;
+import org.kohsuke.stapler.DataBoundSetter;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+
+public class OpenShiftExec extends OpenShiftBaseStep implements IOpenShiftExec {
+
+    protected String pod;
+    protected String container;
+    protected String command;
+    protected List<Argument> arguments = new ArrayList<>();
+    protected String waitTime;
+
+    @DataBoundConstructor
+    public OpenShiftExec( String pod ) {
+        this.pod = pod;
+    }
+
+    @DataBoundSetter
+    public void setContainer(String container) {
+        this.container = container;
+    }
+
+    @DataBoundSetter
+    public void setCommand(String command) {
+        this.command = command;
+    }
+
+    @DataBoundSetter
+    public void setArguments(List<Argument> arguments) {
+        this.arguments = arguments;
+    }
+
+    @DataBoundSetter
+    public void setWaitTime(String waitTime) {
+        this.waitTime = waitTime;
+    }
+
+    public String getPod() {
+        return pod;
+    }
+
+    public String getContainer() {
+        return container;
+    }
+
+    public String getCommand() {
+        return command;
+    }
+
+    public List<Argument> getArguments() {
+        return arguments;
+    }
```

Comment:
Ahh ... separated the command from the arguments, since not strictly tied to java lang constructs here, unlike the restclient changes  ... +1